### PR TITLE
feat(core): always treat pipeline changes as matching that pipeline

### LINKED ISSUE #33: Consider component pipelines to match changes to themselves
i.e. act as if `monorepo: { matches: ".buildkite/pipeline.foo.yml" }` is present in `.buildkite/pipeline.foo.yml`, automatically

### PR DESCRIPTION
**Pull request by dominics**
Fixes #33

### CODE REVIEW

**Review by matthewrj — APPROVED** [review]
How do you test this stuff?

**Comment by dominics** [comment]
It's versioned, so you change the version reference in `.buildkite/generate-pipeline` in-repo (once I get that working - currently choking on `--prefer-offline` because there's no global lockfile and no offline cache configured, so I'll just leave that for a bit)

**Comment by dominics** [comment]
There is a good testing framework in-repo with fixtures and simulated changes, I'll add a test for this PR specifically.

File: src/config.ts [hunk 1]
```diff
@@ -30,14 +30,18 @@ function isConfig(c: ConfigFile): c is Config {
...
 }
 
-async function getConfigFiles(dir: string): Promise<Config[]> {
+/**
+ * Reads all pipeline.foo.yml files from .buildkite/ relative to a given basePath, and returns file locations
+ */
+async function getConfigFiles(basePath: string): Promise<ConfigFile[]> {
   return fs
-    .readdir(dir)
+    .readdir(path.join(basePath, CONFIG_REL_PATH))
```
Comment (dominics) [inline src/config.ts:38]: Reviewed how we load configs, and there's nothing to stop us recursing through all the directories to look for valid pipeline files if we want to support that :+1:

**Pull request merged**